components/butterfly/src/rumor.rs

```diff
@@ -55,6 +58,58 @@ lazy_static! {
                                   &["rumor"]).unwrap();
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Expiration(Option<DateTime<Utc>>);
+
+impl fmt::Display for Expiration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let d = match self.0 {
+            Some(e) => e.to_rfc3339(),
+            None => String::from("None"),
+        };
+
+        write!(f, "{}", d)
+    }
+}
+
+impl Expiration {
+    // This is more of a generic expiration date that we can apply whenever we have a rumor we
+    // don't need to keep around any more.
+    pub fn soon() -> Self { Self(Some(Self::soon_date())) }
+
+    pub fn never() -> Self { Self(None) }
+
+    pub fn new(expiration: DateTime<Utc>) -> Self { Self(Some(expiration)) }
```

Comment:
This can be deleted, since it's never used.

Reply:
It _was_ used at some point.  I'll go back and double check what happened with this.

Reply:
The need for this function disappeared when I implemented `fn expired()` for `Expiration` and I never removed it.  I'll do that.